File: abc/abc145/abc145_b/abc145_b.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
  int n;
  string s;
  cin >> n >> s;
  if(n % 2 == 1) cout << "No";
  else {
    if(s.substr(0, n / 2) == s.substr(n / 2, s.size() - 1)) cout << "Yes";
    else cout << "No";
  }
}

File: abc/abc145/abc145_b/abc145_b.cs
using System;

class A
{
  static void Main()
  {
    Console.ReadLine();
    string s = Console.ReadLine();

    if(s.Length % 2 != 0)
    {
      Console.WriteLine("No");
      return;
    }

    for(int i = 0; i < s.Length / 2; i++)
    {
      if(s[i] != s[i + s.Length / 2])
      {
        Console.WriteLine("No");
        return;
      }
    }

    Console.WriteLine("Yes");
  }
}
